src/emu/natkeyboard.cpp

```diff
@@ -327,6 +327,9 @@ natural_keyboard::natural_keyboard(running_machine &machine)
 	, m_accept_char()
 	, m_charqueue_empty()
 {
+	// special sanity check
+	static_assert(SHIFT_COUNT == UCHAR_SHIFT_END - UCHAR_SHIFT_BEGIN + 1, "SHIFT_COUNT and UCHAR_SHIFT_[BEGIN|END] must line up");
```

Comment:
Since `ioport.h` is in included by `emu.h` and `natkeyboard.h` isn't, why not just change the definition in `natkeyboard.h` to use the `UCHAR_SHIFT_` macros and avoid the `static_assert` here?  That wouldn't introduce an extra implementation dependency in `emu.h` and would marginally simplify things.